tests/Unit/DataStructures/Test_ComplexDataVector.cpp

```diff
@@ -0,0 +1,159 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <complex>
+#include <tuple>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/DereferenceWrapper.hpp"  // IWYU pragma: keep
+#include "Utilities/Functional.hpp"
+#include "Utilities/StdHelpers.hpp"  // IWYU pragma: keep
+#include "tests/Unit/DataStructures/VectorImplTestHelper.hpp"
+
+// IWYU pragma: no_include "DataStructures/DenseMatrix.hpp"
+
+class DataVector;
+
+SPECTRE_TEST_CASE("Unit.DataStructures.ComplexDataVector",
```

Comment:
The same comment as on `ModalVector` PR about combining with the `Math` test and switching from `SECTION` to `INFO`.